src/btree/bt_cursor.c

```diff
@@ -379,6 +381,8 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool *valid)
             }
             if (updp != NULL)
                 *updp = upd;
+            else if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK))
```

Comment:
Doesn't it need the same fix for `BTREE_COL_VAR` page type also?

Reply:
Oh, correct.

Reply:
Fixed